### PR TITLE
autotest-future: changes to the python tester

### PR DESCRIPTION
**Pull request by mishaschwartz**
changes to MarkusPythonTester required to pass settings in json-schema format and automatically update/create tester environments

This PR should be tested in conjunction with https://github.com/MarkUsProject/Markus/pull/3875 on the MarkUs side and with #167 and #168.

### CODE REVIEW

File: testers/testers/py/markus_python_tester.py [hunk 1]
```diff
@@ -141,39 +143,33 @@ def _run_pytest_tests(self, test_file):
...
-                                      else None)
-                for test_file, result in results.items():
-                    for res in result:
-                        test = self.test_class(self, test_file, res, self.feedback_open)
-                        print(test.run(), flush=True)
-        except Exception as e:
-            print(MarkusTester.error_all(message=str(e)), flush=True)
+        results = self.run_python_tests()
+        feedback_file = self.specs.get('test_data', 'feedback_file_name')
+        with MarkusTester.open_feedback(feedback_file) as feedback_open:
```
Comment (adisandro) [inline testers/testers/py/markus_python_tester.py]: Does not belong in this pr, but I think open_feedback can be smarter and save us a bit of typing, if the 'feedback_file_name' setting is always under 'test_data'.
It could be an instance method, using self.specs to fetch the feedback file name itself: `with self.open_feedback as feedback_open:`

File: testers/testers/py/bin/create_environment.sh [hunk 2]
```diff
@@ -3,45 +3,31 @@
...
-    echo "${TESTERDIR}/lib" >> ${pth_file}
-}
-
-get_setting() {
-    echo ${JSONSETTINGS} | python3 -c "import sys, json; print(json.load(sys.stdin)['$1'])"
-}
-
-get_default_setting() {
-	cat ${DEFAULTSETTINGS} | python3 -c "import sys, json; print(json.load(sys.stdin)['$1'])"
+    pip install -r <(echo ${PIP_REQUIREMENTS} | sed 's/\s\+/\n/g')
```
Comment (adisandro) [inline testers/testers/py/bin/create_environment.sh]: Comment what this regexp does.

File: testers/testers/py/bin/create_environment.sh [hunk 3]
```diff
@@ -3,45 +3,31 @@
...
+PIP_REQUIREMENTS="pytest $(echo ${SETTINGS_JSON} | jq --raw-output .env_data.pip_requirements)"
 
-TESTERDIR=$(dirname $(dirname ${THISSCRIPT}))
-VENVDIR=${WORKINGSPECSDIR}/${TESTERNAME}/venv
-PYVERSION=$(get_setting python_version)
-PIPREQUIREMENTS="$(get_default_setting pip_requirements) $(get_setting pip_requirements)"
+VENV_DIR=${ENV_DIR}/venv
+THIS_SCRIPT=$(readlink -f ${BASH_SOURCE})
+LIB_DIR=${THIS_SCRIPT%/*/*}/lib
+TESTERS_DIR=${THIS_SCRIPT%/*/*/*/*}
```
Comment (adisandro) [inline testers/testers/py/bin/create_environment.sh]: I think using dirname or a bunch of readlink -f with THIS_SCRIPT/../../ could be clearer

File: testers/testers/py/markus_python_tester.py (hunk 1 above)
Comment (mishaschwartz) [reply testers/testers/py/markus_python_tester.py]: Agreed, but let's do it in a separate PR once all the PRs for the individual testers have been pulled in so that we can make the change all at once.
